pkg/sdn/plugin/subnets.go

```diff
@@ -75,26 +79,29 @@ func (master *OsdnMaster) addNode(nodeName string, nodeIP string, hsAnnotations
 		}
 	}
 
-	// Create new subnet
-	sn, err := master.subnetAllocator.GetNetwork()
-	if err != nil {
-		return "", fmt.Errorf("error allocating network for node %s: %v", nodeName, err)
-	}
-
-	sub = &osapi.HostSubnet{
-		TypeMeta:   metav1.TypeMeta{Kind: "HostSubnet"},
-		ObjectMeta: metav1.ObjectMeta{Name: nodeName, Annotations: hsAnnotations},
-		Host:       nodeName,
-		HostIP:     nodeIP,
-		Subnet:     sn.String(),
-	}
-	sub, err = master.osClient.HostSubnets().Create(sub)
-	if err != nil {
-		master.subnetAllocator.ReleaseNetwork(sn)
-		return "", fmt.Errorf("error creating subnet %s for node %s: %v", sn.String(), nodeName, err)
+	// Create new subet
+	for _, possibleSubnet := range master.subnetAllocatorList {
+		sn, err := possibleSubnet.GetNetwork()
+		if err == netutils.ErrSubnetAllocatorFull {
+			log.Infof("error allocating network for node %s:%v in subnet %s", nodeName, err)
```

Comment:
I don't think there's any reason to log this unless *all* of the subnets are full. Eg, someone might fill up their original clusterNetwork and then add a second; there's no reason to log a message about the fact that the first network is full every time they add another node, when there's still plenty of room in the second network